services/brig/test/integration/API/Settings.hs

```diff
@@ -0,0 +1,134 @@
+
+                , testCase "EmailVisibleToSelf"
+                . runHttpT manager
+                . withEmailVisibility Opt.EmailVisibleToSelf brig
+                $ testUsersEmailShowsEmailsIfExpected brig galley (expectEmailVisible Opt.EmailVisibleToSelf)
+                ]
+            , testGroup "/users/:id"
+                [ testCase "EmailVisibleIfOnTeam"
+                . runHttpT manager
+                . withEmailVisibility Opt.EmailVisibleIfOnTeam brig
+                $ testGetUserEmailShowsEmailsIfExpected brig galley (expectEmailVisible Opt.EmailVisibleIfOnTeam)
+
+                , testCase "EmailVisibleToSelf"
+                . runHttpT manager
+                . withEmailVisibility Opt.EmailVisibleToSelf brig
+                $ testGetUserEmailShowsEmailsIfExpected brig galley (expectEmailVisible Opt.EmailVisibleToSelf)
+                ]
+            ]
+        ]
+
+data UserRelationship = SameTeam | DifferentTeam | NoTeam
+
+-- Should we show the email for this user type?
+type EmailVisibilityAssertion = UserRelationship -> Bool
+
+expectEmailVisible :: Opt.EmailVisibility -> UserRelationship -> Bool
+expectEmailVisible Opt.EmailVisibleIfOnTeam SameTeam = True
+expectEmailVisible Opt.EmailVisibleIfOnTeam DifferentTeam = True
+expectEmailVisible Opt.EmailVisibleIfOnTeam NoTeam = False
+
+expectEmailVisible Opt.EmailVisibleToSelf SameTeam = False
+expectEmailVisible Opt.EmailVisibleToSelf DifferentTeam = False
+expectEmailVisible Opt.EmailVisibleToSelf NoTeam = False
+
+jsonField :: FromJSON a => Text -> Value -> Maybe a
+jsonField f u = u ^? key f >>= maybeFromJSON
+
+testUsersEmailShowsEmailsIfExpected :: Brig -> Galley -> EmailVisibilityAssertion -> Http ()
+testUsersEmailShowsEmailsIfExpected brig galley shouldShowEmail = do
+    (creatorId, tid) <- createUserWithTeam brig galley
+    (otherTeamCreatorId, otherTid) <- createUserWithTeam brig galley
+    userA <- createTeamMember brig galley creatorId tid Team.fullPermissions
+    userB <- createTeamMember brig galley otherTeamCreatorId otherTid Team.fullPermissions
+    nonTeamUser <- createUser "joe" brig
+    let uids = C8.intercalate "," $ toByteString' <$> [userId userA, userId userB, userId nonTeamUser]
+        expected :: Set (Maybe UserId, Maybe Email)
+        expected = Set.fromList
+                   [ ( Just $ userId userA
+                     , if shouldShowEmail SameTeam then userEmail userA
+                                                   else Nothing)
+                   , ( Just $ userId userB
+                     , if shouldShowEmail DifferentTeam then userEmail userB
+                                                        else Nothing)
+                   , ( Just $ userId nonTeamUser
+                     , if shouldShowEmail NoTeam then userEmail nonTeamUser
+                                                 else Nothing)
+                   ]
+    get (brig . zUser (userId userB) . path "users" . queryItem "ids" uids) !!! do
```

Comment:
```suggestion
    get (brig . zUser (userId userA) . path "users" . queryItem "ids" uids) !!! do
```

it doesn't make a difference with the current definition of `expectEmailVisible`, but it's still good to get this right...